pkg/ingester/ingester_v2.go

```diff
@@ -262,6 +262,38 @@ func (u *userTSDB) getShippedBlocks() ([]ulid.ULID, error) {
 	return shipperMeta.Uploaded, nil
 }
 
+// getOldestUnshippedBlockTime returns the unix timestamp with milliseconds precision of the oldest
+// TSDB block not shipped to the storage yet, or 0 if all blocks have been shipped.
+func (u *userTSDB) getOldestUnshippedBlockTime() (uint64, error) {
+	// Read the list of shipper blocks. If the shipper metadata file doesn't exist
+	// (eg. because the shipper hasn't run yet) we're considering it as if no block
+	// has been shipped.
+	shipped, err := u.getShippedBlocks()
```

Comment:
`getShippedBlocks` is used twice, each time we build a map from it. Perhaps `getShippedBlocks` should build and return map directly?

Reply:
`blocksToDelete()` doesn't build a map of out `getShipperBlocks()`. It just iterates over it.